custom/icds/view_utils.py

```diff
@@ -0,0 +1,25 @@
+from functools import wraps
+
+from django.utils.safestring import mark_safe
+from django.utils.translation import ugettext_lazy
+
+from corehq.apps.hqwebapp.views import no_permissions
+from custom.icds.const import ICDS_DOMAIN, IS_ICDS_ENVIRONMENT
+
+DATA_INTERFACE_ACCESS_DENIED = mark_safe(ugettext_lazy(
+    "This project has blocked access to interfaces that edit data for forms and cases"
+))
+
+
+def check_data_interfaces_blocked_for_domain(view_func):
+    @wraps(view_func)
+    def _inner(request, domain, *args, **kwargs):
+        if data_interfaces_blocked_for_domain(domain):
+            return no_permissions(request, message=DATA_INTERFACE_ACCESS_DENIED)
+        else:
+            return view_func(request, domain, *args, **kwargs)
+    return _inner
+
+
+def data_interfaces_blocked_for_domain(domain):
+    return IS_ICDS_ENVIRONMENT and domain == ICDS_DOMAIN
```

Comment:
we may want to look at making these more configurable via settings since at some point we'll be moving all ICDS specific code to a private repo so you won't be able to import these into the main repo code.